### PR TITLE
GEODE-9146: idle expiration should ignore removed remote entries

### PR DESCRIPTION
**Pull request by dschneider-pivotal**
- sender now tells if expiration is invalid or destroy
- message now checks for removed and invalid entries

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by dschneider-pivotal** [comment]
the dunit failures are known issues

File: geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java [hunk 1]
```diff
@@ -7191,6 +7191,7 @@ void updateStatsForGet(final RegionEntry regionEntry, final boolean hit) {
 
     final long now = cacheTimeMillis();
     if (regionEntry != null) {
+      logger.info("DEBUG: updateStatsForGet now=" + now);
```
Comment (DonalEvans) [inline geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java:7194]: This logging should probably be removed.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PREntryIdleExpirationDistributedTest.java [hunk 2]
```diff
@@ -118,19 +131,22 @@ public void readsInOtherMemberShouldPreventExpiration() throws Exception {
...
       ExpiryTask.permitExpiration();
       Stopwatch stopwatch = Stopwatch.createStarted();
-      while (stopwatch.elapsed(SECONDS) <= 5 && region.containsKey(KEY)) {
+      while (stopwatch.elapsed(SECONDS) <= 3 && region.containsKey(KEY)) {
         Thread.sleep(10);
       }
       assertThat(region.containsKey(KEY)).isTrue();
     });
 
+    LogService.getLogger().info("DEBUG: turning off reads in member3");
```
Comment (DonalEvans) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PREntryIdleExpirationDistributedTest.java:140]: This logging can probably be removed.

**Review by DonalEvans — CHANGES_REQUESTED** [review]
Some left over debug logging that should probably be removed, but otherwise looks good.

File: geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java (hunk 1 above)
Comment (dschneider-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java]: done

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PREntryIdleExpirationDistributedTest.java (hunk 2 above)
Comment (dschneider-pivotal) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PREntryIdleExpirationDistributedTest.java]: done

**Review by DonalEvans — APPROVED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/LatestLastAccessTimeMessage.java [hunk 3]
```diff
@@ -41,17 +43,20 @@
   private int processorId;
   private String regionName;
   private K key;
+  private boolean isInvalidate;
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/internal/cache/LatestLastAccessTimeMessage.java:46]: We could handle this from the receiver point; by not returning the destroy/invalid time to the requester.

**Review by agingade — CHANGES_REQUESTED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/LatestLastAccessTimeMessage.java (hunk 3 above)
Comment (dschneider-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/cache/LatestLastAccessTimeMessage.java]: done

**Review by agingade — APPROVED** [review]

**Review by upthewaterspout — APPROVED** [review]

**Pull request merged**